### PR TITLE
using correct tool mode for `run-make-support` crate

### PR DESCRIPTION
**Pull request by onur-ozkan**
We don't need to ensure std (and rustc) for testing run-make-support's unit tests. Using stage 0 compiler is already enough and speeds up `x test run-make-support` invocations on a clean build.

### CODE REVIEW

File: src/bootstrap/src/core/build_steps/test.rs [hunk 1]
```diff
@@ -678,15 +678,11 @@ impl Step for CompiletestTest {
     /// Runs `cargo test` for compiletest.
     fn run(self, builder: &Builder<'_>) {
         let host = self.host;
-        let compiler = builder.compiler(builder.top_stage, host);
-
-        // We need `ToolStd` for the locally-built sysroot because
-        // compiletest uses unstable features of the `test` crate.
-        builder.ensure(compile::Std::new(compiler, host));
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/build_steps/test.rs:685]: Seems like this was an outdated comment.

**Comment by Kobzol** [comment]
It seems like `compiletest` uses some internal features of `libtest`, and it could perhaps cause issues when `libtest` is changed? :thinking: https://github.com/rust-lang/rust/pull/79001#discussion_r526463035

**Comment by onur-ozkan** [comment]
> It seems like `compiletest` uses some internal features of `libtest`, and it could perhaps cause issues when `libtest` is changed? 🤔 [#79001 (comment)](https://github.com/rust-lang/rust/pull/79001#discussion_r526463035)

Right, that's true. Thanks!

**Comment by onur-ozkan** [comment]
@bors r-

I added explanation comment for compiletest's std usage.

**Comment by workingjubilee** [comment]
There was a failure in https://github.com/rust-lang/rust/pull/127696#issuecomment-2227132583 but it's not clear which PR failed it. Marking all target-specific and CI-related PRs from that rollup as iffy for now.

@bors rollup=iffy

**Pull request merged**